frontends/web/src/components/aopp/aopp.tsx

```diff
@@ -38,164 +37,114 @@ const Banner = ({ children }: TProps) => (
   <div className={styles.banner}>{children}</div>
 );
 
-type State = {
-  accountCode: accountAPI.AccountCode;
-  aopp?: aoppAPI.Aopp;
-}
-
-type Props = TranslateProps;
+const setAccountCodeDefault = (
+  aopp: aoppAPI.Aopp | undefined,
+  setAccountCode: React.Dispatch<React.SetStateAction<string>>
+) => {
+  if (aopp === undefined || aopp.state !== 'choosing-account') {
+    return;
+  }
+  if (aopp.accounts.length) {
+    setAccountCode(aopp.accounts[0].code);
+  }
+};
 
 const domain = (callback: string): string => new URL(callback).host;
 
-class Aopp extends Component<Props, State> {
-  public readonly state: State = {
-    accountCode: '',
-    aopp: undefined,
-  };
-  private unsubscribe?: TUnsubscribe;
-
-  public componentDidMount() {
-    this.setAccountCodeDefault();
-    this.unsubscribe = aoppAPI.subscribeAOPP(aopp => this.updateAOPP(aopp));
-    aoppAPI.getAOPP().then(aopp => this.setState({ aopp }));
-  }
+export const Aopp = () => {
+  const { t } = useTranslation();
 
-  public componentWillUnmount() {
-    if (this.unsubscribe) {
-      this.unsubscribe();
-    }
-  }
+  const [accountCode, setAccountCode] = useState<accountAPI.AccountCode>('');
+  const [aopp, setAopp] = useState<aoppAPI.Aopp>();
 
-  private updateAOPP(aopp: aoppAPI.Aopp) {
-    let shouldUpdateAccountCodeDefault = false;
-    this.setState(currentState => {
-      if (aopp?.state === 'choosing-account'
+  useEffect(() => {
+    setAccountCodeDefault(aopp, setAccountCode);
```

Comment:
looking at the git diff setAccountCodeDefault is confusing me a bit and seems to work differently now.

Reply:
I defined it outside of the component so `aopp` and `setAccountCode` need to be passed to it but it is essentially the same: `aopp` comes from the state so same as [this](https://github.com/NicolaLS/bitbox-wallet-app/blob/3205b4cfde03f303cd76ff586d69fe9f5a1aa1e7/frontends/web/src/components/aopp/aopp.tsx#L89C7-L89C33). and `setAccountCode` sets the `accountCode` state so same as [this](https://github.com/NicolaLS/bitbox-wallet-app/blob/3205b4cfde03f303cd76ff586d69fe9f5a1aa1e7/frontends/web/src/components/aopp/aopp.tsx#L94).

Reply:
I can put it back into the component so it does not need any arguments, I just like the _functional style_ more for some reason but I think it does not really have any impact on anything here (e.g. performance)